agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -51,36 +60,45 @@ def __init__(
             self._running_model_list,
             self._terminal_model,
         )
+        self._problem.nthreads = self._params.nb_threads
+
         # Create solver + callbacks
         self._solver = mim_solvers.SolverCSQP(self._problem)
 
         # Merit function
-        self._solver.use_filter_line_search = self._ocp_params.use_filter_line_search
+        self._solver.use_filter_line_search = self._params.use_filter_line_search
 
         # Parameters of the solver
-        self._solver.termination_tolerance = self._ocp_params.termination_tolerance
-        self._solver.max_qp_iters = self._ocp_params.qp_iters
-        self._solver.eps_abs = self._ocp_params.eps_abs
-        self._solver.eps_rel = self._ocp_params.eps_rel
-        if self._ocp_params.callbacks:
+        self._solver.termination_tolerance = self._params.termination_tolerance
+        self._solver.max_qp_iters = self._params.qp_iters
+        self._solver.eps_abs = self._params.eps_abs
+        self._solver.eps_rel = self._params.eps_rel
+        if self._params.callbacks:
             self._solver.setCallbacks(
                 [mim_solvers.CallbackVerbose(), mim_solvers.CallbackLogger()]
             )
 
     @property
     def horizon_size(self) -> int:
         """Number of time steps in the horizon."""
-        return self._ocp_params.horizon_size
+        return self._params.horizon_size
 
     @property
     def dt(self) -> float:
         """Integration step of the OCP."""
-        return self._ocp_params.dt
+        return self._params.dt
 
     @property
     def problem(self) -> crocoddyl.ShootingProblem:
         return self._problem
 
+    def set_reference_weighted_trajectory(
```

Comment:
shoud we maybe have something like 'debug_mode' flag to only perform this operation in that case?